File: client/src/main/java/commands/pcommands/FilterContainsName.java
package commands.pcommands;

import collection.CollectionFilter;
import commands.*;
import model.data.Model;

import java.util.Arrays;

public final class FilterContainsName extends AbstractCommand {
  private final CollectionFilter collectionFilter;

  public FilterContainsName(CollectionFilter collectionFilter) {
    super(CommandInfo.of("filter_contains_name", "Print elements with specified string in names", true, 1, CommandType.OTHER, ExecutionMode.CLIENT));
    this.collectionFilter = collectionFilter;
  }

  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    String filter = payload.getInlineArg();
    Model[] elements = collectionFilter.filterContainsName(filter);
    String message = Arrays.toString(elements);
    return ExecutionResult.valueOf(true, message);
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/FilterContainsName.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*
import ru.moleus.kollector.domain.collection.CollectionFilter
import java.util.*

class FilterContainsName(private val collectionFilter: CollectionFilter) : AbstractCommand(
    CommandInfo.of(
        "filter_contains_name",
        "Print elements with specified string in names",
        true,
        1,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    override fun execute(payload: ExecutionPayload): ExecutionResult {
        val filter = payload.inlineArg
        val elements = collectionFilter.filterContainsName(filter)
        val message = Arrays.toString(elements)
        return ExecutionResult.valueOf(true, message)
    }
}